app/database/db.go

```diff
@@ -16,6 +16,11 @@ type DB struct {
 	*gorm.DB
 }
 
+// NewDB wraps *gorm.DB
+func NewDB(db *gorm.DB) *DB {
+	return &DB{db}
```

Comment:
Gorm shouldn't be known by the external world, so it should be public (->`newDB` or using struct directly ;-)). The package should change its engine without impact on the other package.

Reply:
fixed